internal/codegen/parse.go

```diff
@@ -0,0 +1,140 @@
+	if err != nil {
+		return nil, err
+	}
+	verificationSchema, ok := verificationSchemaAny.(*openapi3.Schema)
+	if !ok {
+		return nil, errors.New("type assertion failed")
+	}
+	verificationName, ok := verificationSchema.Properties["type"].Value.Enum[0].(string)
+	if !ok {
+		return nil, errors.New("type assertion failed")
+	}
+	verificationName = strings.ReplaceAll(verificationName, "-", "_")
+	docsType, ok := verificationSchema.Extensions["x-docs-type"].(string)
+	if !ok {
+		return nil, errors.New("type assertion failed")
+	}
+
+	// Get verification config data
+	verificationConfigSchemaRef := verificationSchema.Properties["configs"].Ref
+	verificationConfigSchemaRefName := getSchemaNameFromRef(verificationConfigSchemaRef)
+
+	verificationConfigSchemaAny, err := doc.Components.Schemas.JSONLookup(verificationConfigSchemaRefName)
+	if err != nil {
+		return nil, err
+	}
+	verificationConfigSchema, ok := verificationConfigSchemaAny.(*openapi3.Schema)
+	if !ok {
+		return nil, errors.New("type assertion failed")
+	}
+
+	// Constructing Verification struct
+
+	properties := []VerificationProperty{}
+
+	for _, key := range getProperties(verificationName, verificationConfigSchema) {
+		property := verificationConfigSchema.Properties[key]
+		required := slices.Contains(verificationConfigSchema.Required, key)
+		pointerString := ""
+		if !required {
+			pointerString = "Pointer"
+		}
+		typeString := ""
+		if property.Value.Type.Is("string") {
+			typeString = "String"
+		} else if property.Value.Type.Is("number") && property.Value.Format == "float" {
+			typeString = "Float64"
+		}
+		isEnum := false
+		enumNameString := ""
+		if len(property.Value.Enum) > 0 {
+			isEnum = true
+			if property.Ref == "" {
+				enumNameString = "New" + toConfigCase(verificationConfigSchemaRefName) + toConfigCase(key) + "From" + typeString
+			} else {
+				enumNameString = "New" + toConfigCase(getSchemaNameFromRef(property.Ref)) + "From" + typeString
+			}
+		}
+
+		properties = append(properties, VerificationProperty{
+			NameSnake:      key,
+			NameCamel:      strcase.ToLowerCamel(key),
+			NamePascal:     strcase.ToCamel(key),
+			Optional:       !required,
+			Required:       required,
+			Sensitive:      true,
+			PointerString:  pointerString,
+			TypeString:     typeString,
+			IsEnum:         isEnum,
+			EnumNameString: enumNameString,
+		})
+	}
+
+	verification := &Verification{
+		Ref:        refName,
+		ConfigRef:  verificationConfigSchemaRefName,
+		NameSnake:  verificationName,
+		NameCamel:  strcase.ToLowerCamel(verificationName),
+		NamePascal: strcase.ToCamel(verificationName),
+		NameConfig: toConfigCase(docsType),
+		Properties: properties,
+	}
+
+	return verification, nil
+}
+
+func getProperties(verificationName string, schema *openapi3.Schema) []string {
+	if verificationName == "shopify" {
+		// edge case: for Shopify we only care about "webhook_secret_key"
+		return []string{"webhook_secret_key"}
+	}
```

Comment:
I think there's some inconsistency with the Shopify verification schema. From the UI, it only supports "webhook_secret_key" but the API / OpenAPI schema it supports a lot more properties.

Because of some complexity with the typing, I ended up just supporting the secret key only for simplicity sake. Curious to hear your thoughts here. I think there are a few things we can do:

1. Keep it like this
2. Change the schema / API to reflects what we want to support (only webhook secret key, removing the rest)
3. Improve codegen to support the schema (this would also mean the API can do something that the Hookdeck dashboard won't be able to do)

Here's the OpenAPI schema for this

```json
"VerificationShopifyConfigs": {
  "type": "object",
  "properties": {
    "webhook_secret_key": { "type": "string" },
    "rate_limit_period": {
      "type": "string",
      "enum": ["minute", "second"],
      "nullable": true
    },
    "rate_limit": {
      "type": "number",
      "format": "float",
      "nullable": true
    },
    "api_key": { "type": "string", "nullable": true },
    "api_secret": { "type": "string", "nullable": true },
    "shop": { "type": "string", "nullable": true }
  },
  "required": ["webhook_secret_key"],
  "additionalProperties": false,
  "description": "The verification configs for Shopify. Only included if the ?include=verification.configs query param is present",
  "x-docs-type": "ShopifyConfigs"
}
```

Reply:
The schema need to be updated those properties are related to a deprecated feature that was being a feature flag